git-config/src/values.rs

```diff
@@ -273,6 +280,47 @@ impl Serialize for Value<'_> {
     }
 }
 
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[allow(missing_docs)]
+pub struct Path<'a> {
+    pub value: Cow<'a, [u8]>,
+}
+
+impl<'a> TryFrom<Cow<'a, [u8]>> for Path<'a> {
+    type Error = ();
+
+    fn try_from(c: Cow<'a, [u8]>) -> Result<Self, Self::Error> {
+        Self::interpolate(c)
+    }
+}
+
+impl Path<'_> {
+    fn interpolate(val: Cow<'_, [u8]>) -> Result<Path<'_>, ()> {
+        if val.is_empty() {
+            return Err(());
+        }
+
+        if val.starts_with(b"~/") {
```

Comment:
What about adding `const TILDE_PREFIX: &[u8] = b"~/"; `, to allow using its lengths later like so.

```Rust
let (_prefix, val) = val.split_at(TILDE_PREFIX.len());
```